src/components/TodoItem.tsx

```diff
@@ -0,0 +1,202 @@
+import React, {
+  FormEvent,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import classnames from 'classnames';
+import { Todo, Error } from '../types/Index';
+import * as todosService from '../api/todos';
+
+type Props = {
+  todos: Todo[],
+  setTodos: (value: Todo[]) => void,
+  todo: Todo,
+  setHasError: (value: Error) => void,
+  loadingIds: number[],
+  setLoadingIds: React.Dispatch<React.SetStateAction<number[]>>,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todos,
+  setTodos,
+  todo,
+  setHasError,
+  loadingIds,
+  setLoadingIds,
+}) => {
+  const [isEdited, setIsEdited] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  }, [isEdited]);
+
+  const handleDeleteTodoById = () => {
+    setLoadingIds((ids) => {
+      return [...ids, todo.id];
+    });
+    todosService.deleteTodos(todo.id)
+      .then(() => {
+        if (setTodos && todos) {
+          const newTodos: Todo[] = todos.filter(t => t.id !== todo.id);
+
+          setTodos(newTodos);
```

Comment:
better to use callback
```suggestion
          setTodos((prevTodos) => ...);
```